Auto saving drafts

yaelirub opened the pull request:
Using a debouncer to call the saving method every 500ms (defined in constants)

Fixes # #10069

To test:
*Creating draft post*
1. Go to Blog posts
2. Go to Drafts tab
3. Start writing a blog post
4. Rerun XCode (as if the app quit suddenly)
5. Go to Blog Posts
Expected: Your post exists and saved as local with the latest changes.

*Editing post*
1. Go to Blog posts
2. Choose an exiting draft post
3. tap edit
4. start editing 
5.rerun XCode
Expected: Your posts updates exists as local with the latest changes.

nheagy commented on Podfile.lock:
```diff
@@ -266,4 +228,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: d08615d60b668dfbf99e70a4e11e3869879ca70d
 
-COCOAPODS: 1.5.3
+COCOAPODS: 1.3.1
```
You're using an older version of cocoapods. That might work fine, but let's keep the change out of the repo.

nheagy commented on WordPress/WordPress.xcodeproj/project.pbxproj:
```diff
@@ -7151,6 +7156,7 @@
 				7457666E202B558C00F42E40 /* Sources */,
 				7457666F202B558C00F42E40 /* Frameworks */,
 				74576670202B558C00F42E40 /* Resources */,
+				6595B4E51DABEE482E39D7A8 /* [CP] Copy Pods Resources */,
```
some more old cocoapods stuff.

yaelirub commented:
Hey @nheagy , I fixed the crash issue. I was accessing the post managedObjectContext before it was initialized.

I didn't realize we would like to have the same functionality for editing published posts (which makes total sense!) and so now I am using the same debouncing method for editing of a post. I am updating the test plan of this PR to reflect that 

I updated to the latest cocoapod version.

nheagy commented on WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift:
```diff
@@ -483,6 +484,20 @@ class AztecPostViewController: UIViewController, PostEditor {
+                guard let context = strongSelf.post.managedObjectContext else {
+                    return
+                }
+                ContextManager.sharedInstance().save(context)
```
I love that the `!` is gone now!

yaelirub commented on WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift:
```diff
@@ -461,7 +461,8 @@ class AztecPostViewController: UIViewController, PostEditor {
     private var mediaPreviewHelper: MediaPreviewHelper? = nil
 
-
+    // For autosaving
```
Make sense! Will update the PR with your suggestions. Thank you for taking a look!

yaelirub commented:
Thank you guys! 🙏🏻

nheagy commented:
Thanks @yaelirub! 🎉
